### PR TITLE
Additional parsers and expanded others

### PR DESCRIPTION
**Pull request by jhaws1982**

### CODE REVIEW

**Comment by jacketizer** [comment]
Hello, it is a lot of code but I will take a look when I get some time :-)

**Comment by jacketizer** [comment]
> 
> 
> Hello, it is a lot of code but I will take a look when I get some time :-)

I wrote that almost a year ago :-o

**Review by jacketizer — COMMENTED** [review]
Looks good with just some minor comments 👍

**Comment by jhaws1982** [comment]
All of these should have been resolved in the commit just pushed.  Let me know if you have any other concerns.  Thanks!

**Comment by igrr** [comment]
@jhaws1982 could you please rebase the PR on top of the latest master? A few lines have been added in `nmea_date_parse` functions, and the `time` -> `date` renaming needs to be applied to them as well:
https://github.com/jacketizer/libnmea/blob/b0faffa84eddd108bcb4804607e563f6d48d587e/src/parsers/parse.c#L94-L99
This will fix the failing build.

**Comment by jhaws1982** [comment]
@igrr @jacketizer I believe this is good to go now.  Thanks for the feedback and I hope what we are providing is useful!

File: src/parsers/gpvtg.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    case NMEA_GPVTG_TRACKGOOD:
+      data->track_deg = strtod(value, NULL);
+      break;
+    case NMEA_GPVTG_GNDSPD_KNOTS:
+      data->gndspd_knots = strtod(value, NULL);
+      break;
+    case NMEA_GPVTG_GNDSPD_KMPH:
+      data->gndspd_kmph = strtod(value, NULL);
+      break;
+    default:
```
Comment (igrr) [inline src/parsers/gpvtg.c:54]: I'm sorry I didn't notice this the previous time, there is some formating problem around this switch statement (looks like a mix of spaces and tabs)

File: src/parsers/gpgsa.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	return 0;
+}
+
+int
+parse(nmea_parser_s *parser, char *value, int val_index)
+{
+	nmea_gpgsa_s *data = (nmea_gpgsa_s *) parser->data;
+
+	switch (val_index)
+  {
```
Comment (igrr) [inline src/parsers/gpgsa.c:47]: and here as well, please replace leading spaces with tabs

**Review by igrr — CHANGES_REQUESTED** [review]
I'm sorry, just noticed a couple more minor formatting issues — looks like they went unnoticed last time!

File: src/parsers/gpgsa.c (hunk 2 above)
Comment (jhaws1982) [reply src/parsers/gpgsa.c]: Done

File: src/parsers/gpvtg.c (hunk 1 above)
Comment (jhaws1982) [reply src/parsers/gpvtg.c]: Done

**Review by igrr — APPROVED** [review]

**Review by jacketizer — APPROVED** [review]

**Pull request merged**